File: scratch/find.py
import re

with open('app/static/js/dashboard.js', 'r', encoding='utf-8') as f:
    js = f.read()

with open('app/templates/dashboard.html', 'r', encoding='utf-8') as f:
    html = f.read()

matches = re.findall(r"document\.getElementById\(['\"]([^'\"]+)['\"]\)\.innerText\s*=", js)
for m in matches:
    if f'id="{m}"' not in html and f"id='{m}'" not in html:
        print('Missing HTML ID for innerText assignment:', m)

matches = re.findall(r"document\.getElementById\(['\"]([^'\"]+)['\"]\)\.innerHTML\s*=", js)
for m in matches:
    if f'id="{m}"' not in html and f"id='{m}'" not in html:
        print('Missing HTML ID for innerHTML assignment:', m)

File: scratch/find.js
const fs = require('fs');
const js = fs.readFileSync('app/static/js/dashboard.js', 'utf8');
const html = fs.readFileSync('app/templates/dashboard.html', 'utf8');

const regex = /document\.getElementById\(['"]([^'"]+)['"]\)/g;
let match;
let missing = new Set();

while ((match = regex.exec(js)) !== null) {
  const id = match[1];
  if (!html.includes('id="' + id + '"') && !html.includes("id='" + id + "'")) {
    missing.add(id);
  }
}

console.log("Missing elements:", Array.from(missing));
